modules/caddyhttp/matchers.go

```diff
@@ -241,8 +241,16 @@ func (MatchQuery) CaddyModule() caddy.ModuleInfo {
 
 // UnmarshalCaddyfile implements caddyfile.Unmarshaler.
 func (m *MatchQuery) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
+	if *m == nil {
+		*m = make(map[string][]string)
+	}
+
 	for d.Next() {
-		parts := strings.SplitN(d.Val(), "=", 2)
+		var query string
+		if !d.Args(&query) {
+			return d.Errf("malformed query matcher token: %s; must be in param=val format", d.Val())
```

Comment:
A more proper error might be `return d.ArgErr()`, since the argument is missing.